prometheus/registry.go

```diff
@@ -114,6 +114,16 @@ func Register(m Collector) error {
 	return err
 }
 
+// RegisterAll works like Register but takes a variadic list of Collectors.
+func RegisterAll(m ...Collector) error {
+	for i := range m {
+		if err := Register(m[i]); err != nil {
+			return err
```

Comment:
I think that is way to complex to save the user a single for loop. Also, it makes the error handling way more convoluted.